spec/formatting.md

```diff
@@ -106,16 +112,46 @@ In a _pattern_, the resolved value of an _expression_ or _markup_ is used in its
 The form that resolved values take is implementation-dependent,
 and different implementations MAY choose to perform different levels of resolution.
 
-> For example, the resolved value of the _expression_ `{|0.40| :number style=percent}`
-> could be an object such as
+> While this specification does not require it,
+> a _resolved value_ could be implemented by requiring each function implementation to
+> return a value matching the following interface:
 >
+> ```ts
+> interface MessageValue {
+>   formatToString(): string
+>   formatToX(): X // where X is an implementation-defined type
```

Comment:
I may have missed some discussions about this, but why wouldn't we want to be more specific here about the parts?

```suggestion
>   formatToParts(): IterableIterator<MessagePart>; // MessagePart is implementation-defined.
```

Reply:
As this is a non-normative example and we've not reached consensus on formatted parts, I'd rather not introduce them here implicitly and somewhat out of context, when the key aspect of the `formatToX()` inclusion is to remind the reader that a string is not the only possible formatting target.